data_structures/singly_linked_list.py

```diff
@@ -23,26 +17,24 @@ def __init__(self, li):
         p = q
       p.succ = None
     else:
-      self.head = Node(None)
+      self.head = None
 
   def insert_ele(self, val):
-    #insert a element from a sorted singly linked list
-    #first consider if the new node could be inserted as the list head 
-    p = self.head
+    """Insert a element, assuming that this is a sorted singly linked list."""
+    
+    # First consider if the new node could be inserted as the list head.
     q = Node(val)
-    if p.value == None or val < p.value:
-      q.succ = self.head.succ
+    if self.head == None or val < self.head.value:
+      q.succ = self.head
       self.head = q
-    else:
-      #scan through the list until the next value is None or the next value is larger or equal to the one to be inserted
-      while p.succ != None and p.succ.value < val:
-        p = p.succ
-      if p.succ == None:
-        p.succ = q
-        q.succ = None
-      else:
-        q.succ = p.succ
-        p.succ = q
+      return self.head
```

Comment:
After dealing with special cases, do a return. This means that the rest of the function, which deals with "normal" cases, are unindented.